source/agora/network/NetworkClient.d

```diff
@@ -95,21 +95,34 @@ class NetworkClient
 
     /***************************************************************************
 
-        Handshake with the node.
+        Get the node's public key.
 
-        Currently only the node's public key is retrieved,
-        later version checks should be added here.
+        Throws:
+            `Exception` if the request failed.
+
+    ***************************************************************************/
+
+    public void getPublicKey ()
+    {
+        this.key = this.attemptRequest(this.api.getPublicKey(), this.exception);
+    }
+
+    /***************************************************************************
 
-        Note that it is currently blocking until handshake is considered complete.
+        Register the given address as this node's listener
+
+        Params:
+            address = the address to register
 
         Throws:
-            Exception if the handshake did not complete.
+            `Exception` if the request failed.
 
     ***************************************************************************/
 
-    public void handshake ()
+    public void registerListener (string address)
```

Comment:
```suggestion
    public void registerListener (Address address)
```
Can you tell why used string here?

Reply:
Oh I made a mistake while rebasing commits. I'll fix it.

Reply:
Nice catch!